python_modules/dagit/dagit/webapp/src/PipelineExecution.tsx

```diff
@@ -1,21 +1,120 @@
 import * as React from "react";
 import gql from "graphql-tag";
 import styled from "styled-components";
+
+import { ExecutionTab, ExecutionTabs } from "./ExecutionTabs";
 import ConfigCodeEditorContainer from "./configeditor/ConfigCodeEditorContainer";
 import { PipelineExecutionFragment } from "./types/PipelineExecutionFragment";
 import Config from "./Config";
 
+// Internal LocalStorage data format and mutation helpers
+
+interface IStorageData {
```

Comment:
This starts feeling big. Let's separate it to a new file and make a provider (like a React context thingie) that handles storage for this stuff and then wrap this component in it.